bootstrap/cmd/bootstrap/app/ksServer.go

```diff
@@ -1130,28 +1154,29 @@ func timeSinceStart(ctx context.Context) time.Duration {
 	return time.Since(startTime)
 }
 
-func finishDeployment(svc KsService, req CreateRequest, dmDeploy *deploymentmanager.Deployment) {
+func checkDeploymentFinished(svc KsService, req CreateRequest, deployName string) error {
 	status := ""
 	errMsg := ""
 	var err error
 	ctx := context.Background()
 	ctx = context.WithValue(ctx, StartTime, time.Now())
+
 	for retry := 0; retry < 60; retry++ {
 		time.Sleep(10 * time.Second)
-		status, errMsg, err = svc.GetDeploymentStatus(ctx, req)
+		status, errMsg, err = svc.GetDeploymentStatus(ctx, req, deployName)
 		if err != nil {
 			log.Errorf("Failed to get deployment status: %v", err)
 			deployReqCounter.WithLabelValues("INTERNAL").Inc()
 			deploymentFailure.WithLabelValues("INTERNAL").Inc()
-			return
+			return err
 		}
 		if status == "DONE" {
 			if errMsg != "" {
 				log.Errorf("Deployment manager returned error message: %v", errMsg)
 				// Mark status "INVALID_ARGUMENT" as most deployment manager failures are caused by insufficient quota or permission.
 				// Error messages are available from UI, and should be resolvable by retries.
 				deployReqCounter.WithLabelValues("INVALID_ARGUMENT").Inc()
-				return
+				return err
```

Comment:
here ```err == nil```,
but actually we might want to return an error containing errMsg, 
as the deployment action actually failed.

Reply:
done